WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Bulk Edit Price/BulkUpdatePriceSettingsViewModel.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import Yosemite
+
+/// View Model logic for the bulk price setting screen
+///
+final class BulkUpdatePriceSettingsViewModel {
+    /// Represents the possible states for the save button.
+    enum ButtonState: Equatable {
+        case enabled
+        case disabled
+        case loading
+    }
+
+    /// Represents the possible errors during the bulk update
+    enum BulkUpdatePriceError: Error, Equatable {
+        case inputValidationError(ProductPriceSettingsError)
+        case priceUpdateError
+    }
+
+    /// Indicates what price we are editting
+    ///
+    enum EdittingPriceType {
+        case regular
+        case sale
+
+        func keyPathForPriceType() -> KeyPath<ProductVariation, String?> {
+            switch self {
+            case .regular:
+                return \.regularPrice
+            case .sale:
```

Comment:
In now we said to focus on the regular price, but I had already written this beforehand, it does not add much complexity I think so I left it

Reply:
I think we should remove it, otherwise it will be confusing for readers to see code that is not being executed.

Reply:
Ok, I removed it!